js_8/hm10Task2.js

```diff
@@ -1,12 +1,24 @@
 "use strict";
 
-//
-//
-//
-//
-//
-//
-//
-//
-//
-//
\ No newline at end of file
+// У вас есть массив чисел. Напишите функцию countOccurrences, которая принимает массив чисел и возвращает объект
+// с подсчетом каждого числа.
+// Ожидается: { 1: 1, 2: 2, 3: 1, 4: 3, 5: 1}
+
+// const numbers = [1, 2, 2, 3, 4, 4, 4, 5]
+const numbers = [2, 1, 4, 5, 2,  3, 4, 4]
+
+function countOccurrences(arr){
+    arr = arr.toSorted((a,b) => a - b)
+    let count = 1
+    return arr.reduce((acc, element, index, array) => {
+        if (element === array[index + 1]){
+            acc[element] = count++
+        }
+        else {
+            acc[element] = count
+            count = 1
+        }
+        return acc
+    }, {});
```

Comment:
Решено верно, но можно попробовать так
- Для задачки с данньім условием нам не нужно вьіполнять сортировку, она не решает какой-то проблемьі
- `count` тоже не пригодится
- тем самьім немного модифицируем условие в редюсе
```suggestion
  return arr.reduce((accumulator, el) =>
    (accumulator[el]
     ? accumulator[el]++
     : accumulator[el] = 1, accumulator), {});
```